seabird_chat_ingest.proto

```diff
@@ -0,0 +1,144 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+import "common.proto";
+
+package seabird;
+
+// Note that all of the types in this file end with ChatRequest or end with
+// ChatEvent to prevent namespace conflicts between the main seabird.proto file.
+
+// SendMessageChatRequest requests for a message to be sent to a given channel.
+// Either SuccessChatEvent or FailedChatEvent SHOULD be returned in response if
+// possible. This MUST NOT cause any chat events from the bot to appear in the
+// output event stream.
+message SendMessageChatRequest {
+  string channel_id = 1;
+  string message = 2;
+}
+
+// SendPrivateMessageChatRequest requests for a message to be sent to a given
+// user. Either SuccessChatEvent or FailedChatEvent SHOULD be returned in
+// response if possible. This MUST NOT cause any chat events from the bot to
+// appear in the output event stream.
+message SendPrivateMessageChatRequest {
+  string user_id = 1;
+  string message = 2;
+}
+
+// JoinChannelChatRequest requests for the bot to join a channel with the given
+// name. Either JoinChannelEvent or FailedChatEvent SHOULD be returned in
+// response if possible.
+message JoinChannelChatRequest {
+  string channel_name = 1;
+}
+
+// LeaveChannelChatRequest requests for the bot to leave the channel with the
+// given ID. Either LeaveChannelEvent or FailedChatEvent SHOULD be returned in
+// response if possible.
+message LeaveChannelChatRequest {
+  string channel_id = 1;
+}
+
+// UpdateChannelChatRequest requests for a the channel to be updated with the
+// given metadata. Either ChangeChannelEvent or FailedChatEvent SHOULD be returned
+// in response if possible.
+message UpdateChannelChatRequest {
+  string channel_id = 1;
+  string topic = 2;
+}
+
+// Each ChatRequest has an ID (which can be attached to outgoing events - only
+// the first outgoing event with this ID will be used as the reply) and an inner
```

Comment:
This is defined from the perspective of core so should this say `incoming to core` instead of `outgoing`? What happens to other events? Can you explicitly call it out if they simply get dropped?

Reply:
I would also add expectations around what happens when `Request`s don't get a `Response`.